### PR TITLE
Fixes to PUT table configuration API

### PR DESCRIPTION
**Pull request by antumbde**
Handle table not found gracefully and correctly return in case of bad input.
Testing: UT

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java [hunk 1]
```diff
@@ -342,26 +342,35 @@ public Representation updateTableConfig(@Parameter(name = "tableName", in="path"
...
     try {
       String tableTypeStr = config.getTableType();
       TableType tableType = TableType.valueOf(tableTypeStr.toUpperCase());
       String configTableName = config.getTableName();
-      if (! configTableName.equals(tableName)) {
-        errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST,
+      String requestTablNameWithType = new TableNameBuilder(tableType).forTable(tableName);
+      if (! configTableName.equals(requestTablNameWithType)) {
+        return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST,
             "Request table name does not match table name in the body");
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java]: please add the name of the table in the config in the message.

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java [hunk 2]
```diff
@@ -207,6 +209,17 @@ public void testUpdateTableConfig()
     sendPutRequest(controllerUrlBuilder.forUpdateTableConfig(tableName), tableConfig.toJSON().toString());
     modifiedConfig = getTableConfig(tableName, "REALTIME");
     Assert.assertEquals(modifiedConfig.getQuotaConfig().getStorage(), "10G");
+    try {
+      // table does not exist
+      JSONObject jsonConfig = tableConfig.toJSON();
+      jsonConfig.put("tableName", "noSuchTable_REALTIME");
+      String responseStr =
+          sendPutRequest(controllerUrlBuilder.forUpdateTableConfig("noSuchTable"), jsonConfig.toString());
+      System.out.println(responseStr);
```
Comment (mcvsubbu) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java]: remove println

Comment (mcvsubbu) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java]: we should compare this with some error msg? Or, at least validate that this is an error (as opposed to success)?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java (hunk 1 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java]: Added. But this is response so user has the information

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java (hunk 2 above)
Comment (antumbde) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java]: I am validating for error in the catch block

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 3]
```diff
@@ -1097,17 +1097,16 @@ private void createHelixEntriesForHighLevelConsumer(AbstractTableConfig config,
...
+      throws IOException {
     if (type == TableType.REALTIME) {
       ZKMetadataProvider.setRealtimeTableConfig(_propertyStore, tableNameWithSuffix,
           AbstractTableConfig.toZnRecord(config));
       ensureRealtimeClusterIsSetUp(config, tableNameWithSuffix, config.getIndexingConfig());
-    } else {
+    } else if (type == TableType.OFFLINE) {
       ZKMetadataProvider.setOfflineTableConfig(_propertyStore, tableNameWithSuffix,
           AbstractTableConfig.toZnRecord(config));
     }
```
Comment (jfim) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java]: Should throw an exception if it's neither realtime/offline

Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java]: I'll add that. Submitting this one to make it part of release

Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java]: Just checked. AbstractTableConfig.init() handles this

**Pull request merged**